File: ruby/1_two_sum.rb
#!/usr/env ruby
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  nums.combination(2) do |a, b| 
    sum = a + b
    return [nums.index(a), nums.rindex(b)] if sum == target
  end
end

two_sum([1,2,3,4,5], 9)

File: typescript/1_two_sum.ts
function twoSum(nums: number[], target: number): number[] {
  const ans = [] as number[]

  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] == target) {
        ans.push(i)
        ans.push(j)
      }
    }
  }

  return ans
}

const ret = twoSum([2, 7, 11, 15], 9)
const exp = [0, 1]

if (JSON.stringify(ret) == JSON.stringify(exp)) {
  console.log('ok')
}
